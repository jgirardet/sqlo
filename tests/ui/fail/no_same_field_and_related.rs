#[derive(sqlo::Sqlo)]
struct Bla {
    id: i32,
    un: String,
}

#[derive(sqlo::Sqlo)]
struct Bli {
    id: i32,
    #[sqlo(fk = "Bla", related = "un")]
    deux: i32,
}

fn main() {}
// the error message has many error, the important one is about related and field name
