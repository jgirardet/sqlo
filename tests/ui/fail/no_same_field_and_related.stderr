error: error returned from database: (code: 1) no such table: bla
 --> tests/ui/fail/no_same_field_and_related.rs:1:10
  |
1 | #[derive(sqlo::Sqlo)]
  |          ^^^^^^^^^^
  |
  = note: this error originates in the macro `$crate::sqlx_macros::expand_query` which comes from the expansion of the derive macro `sqlo::Sqlo` (in Nightly builds, run with -Z macro-backtrace for more info)

error: error returned from database: (code: 1) no such table: bla
 --> tests/ui/fail/no_same_field_and_related.rs:1:10
  |
1 | #[derive(sqlo::Sqlo)]
  |          ^^^^^^^^^^
  |
  = note: this error originates in the macro `$crate::sqlx_macros::expand_query` which comes from the expansion of the derive macro `sqlo::Sqlo` (in Nightly builds, run with -Z macro-backtrace for more info)

error: error returned from database: (code: 1) no such table: bla
 --> tests/ui/fail/no_same_field_and_related.rs:1:10
  |
1 | #[derive(sqlo::Sqlo)]
  |          ^^^^^^^^^^
  |
  = note: this error originates in the macro `$crate::sqlx_macros::expand_query` which comes from the expansion of the derive macro `sqlo::Sqlo` (in Nightly builds, run with -Z macro-backtrace for more info)

error: error returned from database: (code: 1) no such table: bla
 --> tests/ui/fail/no_same_field_and_related.rs:1:10
  |
1 | #[derive(sqlo::Sqlo)]
  |          ^^^^^^^^^^
  |
  = note: this error originates in the macro `$crate::sqlx_macros::expand_query` which comes from the expansion of the derive macro `sqlo::Sqlo` (in Nightly builds, run with -Z macro-backtrace for more info)

error: related name must be different from all targeted sqlos's fields
  --> tests/ui/fail/no_same_field_and_related.rs:10:34
   |
10 |     #[sqlo(fk = "Bla", related = "un")]
   |                                  ^^^^
